### PR TITLE
test: Code coverage View Mileage

### PR DESCRIPTION
**Pull request by CODEX108**
clickUp: https://app.clickup.com/t/85zt05a2f
Screenshot 2023-06-01 at 3 20 55 PM

### CODE REVIEW

File: src/app/fyle/view-mileage/view-mileage.page.spec.ts [hunk 1]
```diff
@@ -18,14 +18,21 @@ import { PopoverController, ModalController } from '@ionic/angular';
 import { ExpenseView } from 'src/app/core/models/expense-view.enum';
 import { EventEmitter } from '@angular/core';
 import { of } from 'rxjs';
-import { expenseData1, expenseData2 } from 'src/app/core/mock-data/expense.data';
+import { etxncData, expenseData1, expenseData2 } from 'src/app/core/mock-data/expense.data';
```
Comment (yksolanki9) [inline src/app/fyle/view-mileage/view-mileage.page.spec.ts:21]: Rename to etxnData?

File: src/app/fyle/view-mileage/view-mileage.page.spec.ts [hunk 2]
```diff
@@ -459,6 +466,482 @@ describe('ViewMileagePage', () => {
...
+      component.extendedMileage$ = of(etxncData.data[0]);
+      expenseFieldsService.getAllMap.and.returnValue(of(expenseFieldsMapResponse));
+      component.txnFields$ = of(expenseFieldsMapResponse);
+      orgSettingsService.get.and.returnValue(of(mockOrgSettData));
+      component.ionViewWillEnter();
+      tick(500);
+      expect(component.isProjectShown).toBeFalsy();
+    }));
+
+    it('should get all the org setting and return true if new reports Flow Enabled ', () => {
```
Comment (yksolanki9) [inline src/app/fyle/view-mileage/view-mileage.page.spec.ts:700]: Write one test case for when projectFieldName is present, project is not mandatory and tx_project_name is set to some value.

File: src/app/fyle/view-mileage/view-mileage.page.spec.ts [hunk 3]
```diff
@@ -459,6 +466,482 @@ describe('ViewMileagePage', () => {
...
+      component.txnFields$ = of(expenseFieldsMapResponse4);
+      activateRouteMock.snapshot.params.view = ExpenseView.team;
+
+      component.ionViewWillEnter();
+      component.canDelete$.subscribe((res) => {
+        expect(mockExtMileageData.tx_state).toEqual('PAID');
+        expect(res).toBeFalse();
+        done();
+      });
+    });
```
Comment (yksolanki9) [inline src/app/fyle/view-mileage/view-mileage.page.spec.ts:781]: Set view = Individual and check that canFlagorUnflag and canDelete should be false

File: src/app/fyle/view-mileage/view-mileage.page.spec.ts [hunk 4]
```diff
@@ -459,6 +466,482 @@ describe('ViewMileagePage', () => {
...
+      activateRouteMock.snapshot.params = {
+        id: 'tx3qwe4ty',
+        view: ExpenseView.individual,
+        txnIds: '["tx3qwe4ty","tx6sd7gh","txD3cvb6"]',
+        activeIndex: '20',
+      };
+      component.ionViewWillEnter();
+      expect(component.updateFlag$.next).toHaveBeenCalledOnceWith(null);
+      expect(component.numEtxnsInReport).toEqual(3);
+      expect(component.activeEtxnIndex).toEqual(20);
```
Comment (yksolanki9) [inline src/app/fyle/view-mileage/view-mileage.page.spec.ts:941]: Logicslly incorrect -  If there are 3 txns in report then the activeIndex cannot be 20. Change it to 2

**Review by yksolanki9 — APPROVED** [review]
Minor comments

File: src/app/fyle/view-mileage/view-mileage.page.spec.ts (hunk 4 above)
Comment (yksolanki9) [reply src/app/fyle/view-mileage/view-mileage.page.spec.ts:998]: The ask was to change the activeIndex, not the number of txns in report

**Review by yksolanki9 — CHANGES_REQUESTED** [review]

**Review by Dimple16 — COMMENTED** [review]
resolve existing comments

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**